test/test_nrm2_device.cc

```diff
@@ -0,0 +1,183 @@
+    real_t* result = &result_host;
+    real_t  result_cblas;
+
+    // mark non-standard output values
+    params.gflops();
+    params.gbytes();
+    params.ref_time();
+    params.ref_gflops();
+    params.ref_gbytes();
+
+    // adjust header to msec
+    params.time.name( "BLAS++\ntime (ms)" );
+    params.ref_time.name( "Ref.\ntime (ms)" );
+
+    if (! run)
+        return;
+
+    if (blas::get_device_count() == 0) {
+        params.msg() = "skipping: no GPU devices or no GPU support";
+        return;
+    }
+
+    // setup
+    size_t size_x = (n - 1) * std::abs(incx) + 1;
+    Tx* x    = new Tx[ size_x ];
+    Tx* xref = new Tx[ size_x ];
+
+    // device specifics
+    blas::Queue queue(device, 0);
+    Tx* dx;
+
+    dx = blas::device_malloc<Tx>(size_x, queue);
+    if (mode == 'd') {
+        result = blas::device_malloc<real_t>(1, queue);
+        #if defined( BLAS_HAVE_CUBLAS )
+        cublasSetPointerMode(queue.handle(), CUBLAS_POINTER_MODE_DEVICE);
+        #elif defined( BLAS_HAVE_ROCBLAS )
+        rocblas_set_pointer_mode( queue.handle(), rocblas_pointer_mode_device );
+        #endif
+    }
+
+    // test error exits
+    assert_throw( blas::nrm2( -1, x, incx, result, queue ), blas::Error );
+    assert_throw( blas::nrm2(  n, x,    0, result, queue ), blas::Error );
+
+    int64_t idist = 1;
+    int iseed[4] = { 0, 0, 0, 1 };
+    lapack_larnv( idist, iseed, size_x, x );
+    cblas_copy( n, x, incx, xref, incx );
+
+    blas::device_setvector(n, x, std::abs(incx), dx, std::abs(incx), queue);
+    queue.sync();
+
+    if (verbose >= 1) {
+        printf( "\n"
+                "n=%5lld, incx=%5lld, sizex=%10lld\n",
+                (lld) n, (lld) incx, (lld) size_x );
+    }
+    if (verbose >= 2) {
+        printf( "x    = " ); print_vector( n, x, incx );
+    }
+
+    // run test
+    testsweeper::flush_cache( params.cache() );
+    double time = get_wtime();
+    blas::nrm2( n, dx, incx, result, queue );
+    queue.sync();
+    time = get_wtime() - time;
+
+    if (mode == 'd') {
+        device_memcpy( &result_host, result, 1, queue );
+    }
+
+    double gflop = Gflop<Tx>::nrm2( n );
+    double gbyte = Gbyte<Tx>::nrm2( n );
+    params.time()   = time * 1000;  // msec
+    params.gflops() = gflop / time;
+    params.gbytes() = gbyte / time;
+
+    blas::device_getvector(n, dx, std::abs(incx), x, std::abs(incx), queue);
+    queue.sync();
+
+    if (verbose >= 2) {
+        printf( "x2   = " ); print_vector( n, x, incx );
+    }
+
+    if (params.check() == 'y') {
+        // run reference
+        testsweeper::flush_cache( params.cache() );
+        time = get_wtime();
+        result_cblas = cblas_nrm2( n, xref, incx );
+        time = get_wtime() - time;
+
+        params.ref_time()   = time * 1000;  // msec
+        params.ref_gflops() = gflop / time;
+        params.ref_gbytes() = gbyte / time;
+
+        if (verbose >= 2) {
+            printf( "result0 = %3.2lld\n", (lld) result_cblas);
```

Comment:
This should be floating point, something like `%.2e` or `%.3g` or `%.2f`, not `%lld` with `(lld)` cast.

I fixed it in switching to `llong` instead of `lld`.